File: dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  bool isArmstrongNumber(int n) {
    // convert to string and get length
    String digits = n.toString();
    int length = digits.length;

    // split into list and fold into sum
    int sum = digits.split("")
      .fold(0, (sum, char) => sum + pow(int.parse(char), length).round());

    return sum == n;
  }
}

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        string numberString = number.ToString();
        return (int)numberString.Sum(c => Math.Pow(Char.GetNumericValue(c), numberString.Length)) == number;
    }
}